lldb/source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -135,79 +135,98 @@ static bool HasReflectionInfo(ObjectFile *obj_file) {
   return hasReflectionSection;
 }
 
+SwiftLanguageRuntime::NativeReflectionContext *
+SwiftLanguageRuntime::GetReflectionContext() {
+  if (!m_initialized_reflection_ctx)
+    SetupReflection();
+  return m_reflection_ctx.get();
+}
+
 void SwiftLanguageRuntime::SetupReflection() {
-  reflection_ctx.reset(new NativeReflectionContext(this->GetMemoryReader()));
+  std::lock_guard<std::recursive_mutex> lock(m_add_module_mutex);
+
```

Comment:
I think you need to check that m_initialized_reflection_ctx is false here before continuing.  If two threads call GetReflectionContext at the same time, before it is set up, they could both get as far as locking m_add_module_mutex.  Then when the first one is done initializing, and drops the lock, the second one will do it again.